proxy/src/console/provider/neon.rs

```diff
@@ -300,23 +312,61 @@ impl super::Api for Api {
             .wake_compute_endpoint_rate_limiter
             .check(user_info.endpoint.normalize_intern(), 1)
         {
-            info!(key = &*key, "found cached compute node info");
             return Err(WakeComputeError::TooManyConnections);
         }
 
-        let mut node = permit.release_result(self.do_wake_compute(ctx, user_info).await)?;
-        ctx.set_project(node.aux.clone());
-        let cold_start_info = node.aux.cold_start_info;
-        info!("woken up a compute node");
+        let node = permit.release_result(self.do_wake_compute(ctx, user_info).await);
+        match node {
+            Ok(mut node) => {
+                ctx.set_project(node.aux.clone());
+                let cold_start_info = node.aux.cold_start_info;
+                debug!("woken up a compute node");
+
+                // store the cached node as 'warm'
+                node.aux.cold_start_info = ColdStartInfo::WarmCached;
+                let (_, cached) = self
+                    .caches
+                    .node_info
+                    .insert_unit(key.clone(), Ok(node.clone()));
+                node.aux.cold_start_info = cold_start_info;
+
+                debug!(key = &*key, "created a cache entry for compute node info");
+
+                Ok(cached.map(|()| node))
+            }
+            Err(err) => match err {
+                WakeComputeError::ApiError(ApiError::Console(err)) => {
+                    let Some(status) = &err.status else {
+                        return Err(WakeComputeError::ApiError(ApiError::Console(err)));
+                    };
+
+                    let reason = status
+                        .details
+                        .error_info
+                        .map_or(Reason::Unknown, |x| x.reason);
+
+                    // if we can retry this error, do not cache it.
+                    if reason.can_retry() {
+                        return Err(WakeComputeError::ApiError(ApiError::Console(err)));
+                    }
 
-        // store the cached node as 'warm'
-        node.aux.cold_start_info = ColdStartInfo::WarmCached;
-        let (_, mut cached) = self.caches.node_info.insert(key.clone(), node);
-        cached.aux.cold_start_info = cold_start_info;
+                    // at this point, we should only have quota errors.
+                    debug!(
+                        key = &*key,
+                        "created a cache entry for the wake compute error"
```

Comment:
Should this state that it is a quota error, actually?

Reply:
Not quite sure what benefit that would be

Reply:
Help debugging? :) The comment says that this is a quote error. So if we look at the logs, we could see this additional information. But not important.